source/extensions/filters/network/postgres_proxy/postgres_decoder.cc

```diff
@@ -30,11 +32,11 @@ void DecoderImpl::initialize() {
   FE_known_msgs['p'] =
       MsgProcessor{"PasswordMessage/GSSResponse/SASLInitialResponse/SASLResponse", {}};
   FE_known_msgs['P'] = MsgProcessor{"Parse", {}};
-  FE_known_msgs['Q'] = MsgProcessor{"Query", {}};
+  FE_known_msgs['Q'] = MsgProcessor{"Query", {&DecoderImpl::onQuery}};
```

Comment:
@cpakulski I found a bug here because processing `Q` frontend command will cover just `Simple` protocol. To cover `Extended` protocol you should call `onQuery` also for `P` frontend command. Something like below:

```diff
diff --git a/source/extensions/filters/network/postgres_proxy/postgres_decoder.cc b/source/extensions/filters/network/postgres_proxy/postgres_decoder.cc
index 90c4edb05..0ba94ce54 100644
--- a/source/extensions/filters/network/postgres_proxy/postgres_decoder.cc
+++ b/source/extensions/filters/network/postgres_proxy/postgres_decoder.cc
@@ -31,7 +31,7 @@ void DecoderImpl::initialize() {
   FE_known_msgs['F'] = MsgProcessor{"FunctionCall", {}};
   FE_known_msgs['p'] =
       MsgProcessor{"PasswordMessage/GSSResponse/SASLInitialResponse/SASLResponse", {}};
-  FE_known_msgs['P'] = MsgProcessor{"Parse", {}};
+  FE_known_msgs['P'] = MsgProcessor{"Parse", {&DecoderImpl::onQuery}};
   FE_known_msgs['Q'] = MsgProcessor{"Query", {&DecoderImpl::onQuery}};
   FE_known_msgs['S'] = MsgProcessor{"Sync", {}};
   FE_known_msgs['X'] = MsgProcessor{"Terminate", {&DecoderImpl::decodeFrontendTerminate}};
```